twoliter/embedded/rpm2img

```diff
@@ -251,128 +249,20 @@ mv "${ROOT_MOUNT}/boot/efi"/* "${EFI_MOUNT}"
-      -out "${SBKEYS}/shim-sign.p12"
-
-    openssl pkcs12 \
-      -export \
-      -passout pass: \
-      -inkey "${SBKEYS}/code-sign.key" \
-      -in "${SBKEYS}/code-sign.crt" \
-      -certfile "${SBKEYS}/vendor.crt" \
-      -out "${SBKEYS}/code-sign.p12"
-
-    # Import certificates and private key archive.
-    PEDB="/etc/pki/pesign"
-
-    certutil -d "${PEDB}" -A -n db -i "${SBKEYS}/db.crt" -t ",,C"
-    certutil -d "${PEDB}" -A -n shim-sign-key -i "${SBKEYS}/shim-sign.crt" -t ",,P"
-    pk12util -d "${PEDB}" -i "${SBKEYS}/shim-sign.p12" -W ""
-
-    certutil -d "${PEDB}" -A -n vendor -i "${SBKEYS}/vendor.crt" -t ",,C"
-    certutil -d "${PEDB}" -A -n code-sign-key -i "${SBKEYS}/code-sign.crt" -t ",,P"
-    pk12util -d "${PEDB}" -i "${SBKEYS}/code-sign.p12" -W ""
-
-    certutil -d "${PEDB}" -L
-    SB_KEY_SOURCE="local"
-    SHIM_SIGN_KEY=(-c shim-sign-key)
-    CODE_SIGN_KEY=(-c code-sign-key)
-  fi
-
-  # Convert certificates from PEM format (ASCII) to DER (binary). This could be
-  # done when the certificates are created, but the resulting binary files are
-  # not as nice to store in source control.
-  for cert in PK KEK db vendor ; do
-    openssl x509 \
-      -inform PEM -in "${SBKEYS}/${cert}.crt" \
-      -outform DER -out "${SBKEYS}/${cert}.cer"
-  done
-
-  # For signing the grub config, we need to embed the GPG public key in binary
-  # form, which is similarly awkward to store in source control.
-  gpg --import "${SBKEYS}/config-sign.key"
-  if [ "${SB_KEY_SOURCE}" == "aws" ] ; then
-    gpg --card-status
+    sbsetup_local_profile SHIM_SIGN_KEY CODE_SIGN_KEY
   fi
-  gpg --export > "${SBKEYS}/config-sign.pubkey"
-  gpg --list-keys
 fi
 
-pushd "${EFI_MOUNT}/EFI/BOOT" >/dev/null
-shims=(boot*.efi)
-shim="${shims[0]}"
-grubs=(grub*.efi)
-grub="${grubs[0]}"
-mokms=(mm*.efi)
-mokm="${mokms[0]}"
 if [ "${UEFI_SECURE_BOOT}" == "yes" ] ; then
-  # Convert the vendor certificate to the expected format.
-  cert_table "${SBKEYS}/vendor.cer" "${SBKEYS}/vendor.obj"
-
-  # Replace the embedded vendor certificate, then sign shim with the db key.
-  "${ARCH}-objdumpcopy" "${shim}" \
-    --update-section ".vendor_cert=${SBKEYS}/vendor.obj"
-  pesign -i "${shim}" -o "${shim}.signed" -s "${SHIM_SIGN_KEY[@]}"
-  mv "${shim}.signed" "${shim}"
-  pesigcheck -i "${shim}" -n 0 -c "${SBKEYS}/db.cer"
-
-  # Sign the MOK manager as well.
-  pesign -i "${mokm}" -o "${mokm}.signed" -s "${CODE_SIGN_KEY[@]}"
-  mv "${mokm}.signed" "${mokm}"
-  pesigcheck -i "${mokm}" -n 0 -c "${SBKEYS}/vendor.cer"
-
-  # Replace the embedded gpg public key, then sign grub with the vendor key.
-  "${ARCH}-objdumpcopy" "${grub}" \
-    --file-alignment 4096 \
-    --update-section ".pubkey=${SBKEYS}/config-sign.pubkey"
-  pesign -i "${grub}" -o "${grub}.signed" -s "${CODE_SIGN_KEY[@]}"
-  mv "${grub}.signed" "${grub}"
-  pesigcheck -i "${grub}" -n 0 -c "${SBKEYS}/vendor.cer"
+  uefi_pesign "${EFI_MOUNT}" "${ARCH}" SHIM_SIGN_KEY CODE_SIGN_KEY
```

Comment:
You should refactor this along the lines of `<sign_function> <ARTIFACT>` for consistency with the other signing functions.

You are also not very consistent about when you pass in keys (here) versus when you have the function decide what key to use (`sign_grub`). I would recommend just specializing all of the functions to sign a particular artifact and have the function pick the key to use. There is only one correct choice for a given artifact - using the wrong key will either result in an unbootable image or weaken security.